src/imu_control_parameters.cpp

```diff
@@ -0,0 +1,331 @@
+    m_funcMap["anglvel_calibbias_z"] = &IIOWrapper::update_anglvel_calibbias_z;
+    m_funcMap["sampling_frequency"] = &IIOWrapper::update_sampling_frequency;
+
+    m_funcMap2["software_reset"] = &IIOWrapper::software_reset;
+    m_funcMap2["flash_memory_test"] = &IIOWrapper::flash_memory_test;
+    m_funcMap2["flash_memory_update"] = &IIOWrapper::flash_memory_update;
+    m_funcMap2["sensor_self_test"] = &IIOWrapper::sensor_self_test;
+    m_funcMap2["factory_calibration_restore"] = &IIOWrapper::factory_calibration_restore;
+
+    m_funcMapGet["filter_size"] = &IIOWrapper::filter_size;
+    m_funcMapGet["burst_size_selection"] = &IIOWrapper::burst_size_selection;
+    m_funcMapGet["burst_data_selection"] = &IIOWrapper::burst_data_selection;
+    m_funcMapGet["linear_acceleration_compensation"] = &IIOWrapper::linear_acceleration_compensation;
+    m_funcMapGet["point_of_percussion_alignment"] = &IIOWrapper::point_of_percussion_alignment;
+    m_funcMapGet["internal_sensor_bandwidth"] = &IIOWrapper::internal_sensor_bandwidth;
+    m_funcMapGet["sync_mode_select"] = &IIOWrapper::sync_mode_select;
+    m_funcMapGet["sync_polarity"] = &IIOWrapper::sync_polarity;
+    m_funcMapGet["data_ready_polarity"] = &IIOWrapper::data_ready_polarity;
+    m_funcMapGet["sync_signal_scale"] = &IIOWrapper::sync_signal_scale;
+    m_funcMapGet["decimation_filter"] = &IIOWrapper::decimation_filter;
+    m_funcMapGet["accel_calibbias_x"] = &IIOWrapper::accel_x_calibbias;
+    m_funcMapGet["accel_calibbias_y"] = &IIOWrapper::accel_y_calibbias;
+    m_funcMapGet["accel_calibbias_z"] = &IIOWrapper::accel_z_calibbias;
+    m_funcMapGet["anglvel_calibbias_x"] = &IIOWrapper::anglvel_x_calibbias;
+    m_funcMapGet["anglvel_calibbias_y"] = &IIOWrapper::anglvel_y_calibbias;
+    m_funcMapGet["anglvel_calibbias_z"] = &IIOWrapper::anglvel_z_calibbias;
+    m_funcMapGet["sampling_frequency"] = &IIOWrapper::sampling_frequency;
+}
+
+bool ImuControlParameters::setParameters(std::map<std::string, int32_t>& valueMap)
+{
+    bool hasError = false;
+
+    std::vector<std::string> keys;
+    for (auto it = m_funcMap.begin(); it != m_funcMap.end(); it++) {
+        keys.push_back(it->first);
+    }
+
+    for (const std::string &key: keys)
+    {
+        int deviceParamValue = (m_iioWrapper.*(m_funcMapGet[key]))();
+
+        if(deviceParamValue != valueMap[key])
+        {
+            const char * ckey = key.c_str();
+
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_parameter_before_set"), "%s: '%d'", ckey , deviceParamValue);
+
+            int ret =(m_iioWrapper.*(m_funcMap[key]))(valueMap[key]);
+
+            if(ret == 0)
+            {
+                RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "%s: '%d'", ckey , valueMap[key]);
+            }
+            else
+            {
+                RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter_error"), "error on set %s: '%d' , iio_device_debug_attr_write_longlong:%d", ckey , valueMap[key], ret);
```

Comment:
remove iio_device_debug_attr_write_longlong. simply state that the parameter setting was not successful